### ISSUE TITLE
Extra "void" parameter in C++ implementation

### ISSUE THREAD

**Issue by firemansamm** [hnwangkg-ezio/LCD1in8#1]
Having the extra `void` in the method signature of C++ init function causes compilation to fail.

On this line:
```
void LCD_Init(void){
```

Running `pxt test` to compile a binary with the package fails with:
```
Error: argument number mismatch: 0 vs 1 in C++
    at Object.userError (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:54789:25)
    at Object.validateShim (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:71953:32)
    at emitFunctionDeclaration (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:69497:34)
    at emitNodeCore (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70399:25)
    at catchErrors (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70342:33)
    at emit (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70381:17)
    at Array.forEach (<anonymous>)
    at emitBlock (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:69944:33)
    at emitNodeCore (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70405:32)
    at catchErrors (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70342:33)
    at emit (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70381:17)
    at emitModuleDeclaration (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70328:17)
    at emitNodeCore (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70392:32)
    at catchErrors (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70342:33)
    at emit (C:\Users\Sam\Desktop\testing\node_modules\pxt-core\built\pxt.js:70381:17)
    at Array.forEach (<anonymous>)
LCD1in8.ts(25,21): error TS9200: argument number mismatch: 0 vs 1 in C++
```

I will submit a PR shortly resolving this.

**Comment by hnwangkg-ezio** [comment]
Using target PXT/microbit with build engine yotta
  Target dir:   /home/ezio/microbit/node_modules/pxt-microbit
  PXT Core dir: /home/ezio/microbit/node_modules/pxt-core
*** yotta target bbc-microbit-classic-gcc
INTERNAL ERROR: Error: spawn yotta ENOENT
    at exports._errnoException (util.js:870:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:441:9)
    at process._tickCallback (node.js:355:17)
这是需要我装yotta交叉编译工具链吗？

**Comment by firemansamm** [comment]
对 这是因为pxt找不到yotta。使用pxt build跟pxt test可验证c++ shims的正常运作。

**Comment by hnwangkg-ezio** [comment]
十分感谢，总算遇到国内大牛了。不过这个yotta我安装的时候提示
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-w2BjnN/semantic-version/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-Rlxkpz-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-w2BjnN/semantic-version/
这个，这个是什么问题导致？

**Comment by hnwangkg-ezio** [comment]
我加了sudo,好像是安装成功了。

**Comment by hnwangkg-ezio** [comment]
ezio@ubuntu:~/microbit/LED$ pxt build
Using target PXT/microbit with build engine yotta
  Target dir:   /home/ezio/microbit/node_modules/pxt-microbit
  PXT Core dir: /home/ezio/microbit/node_modules/pxt-core
*** yotta build
info: generate for target: bbc-microbit-classic-gcc 0.2.3 at /home/ezio/microbit/LED/built/yt/yotta_targets/bbc-microbit-classic-gcc
GCC version is: 7.2.1
suppressing warnings from ble-nrf51822
suppressing warnings from nrf51-sdk
suppressing ALL warnings from mbed-classic, ble, ble-nrf51822 & nrf51-sdk
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ezio/microbit/LED/built/yt/build/bbc-microbit-classic-gcc
[1/1] Linking CXX executable source/pxt-microbit-app
Package built; written to binary.hex; size: 579207
ezio@ubuntu:~/microbit/LED$ pxt test
Using target PXT/microbit with build engine yotta
  Target dir:   /home/ezio/microbit/node_modules/pxt-microbit
  PXT Core dir: /home/ezio/microbit/node_modules/pxt-core
Package built; written to binary.hex; size: 579207


这样是算成功了吗？

**Comment by firemansamm** [comment]
是的 唯一能完全验证配件的运作是把使用package的代码加入tests.ts,并把hex上传到microbit。不过只要pxt test成功的话，代表编译应该没为题。

**Comment by hnwangkg-ezio** [comment]
好的，谢谢您。这个问题我解决了。现在测试的起来比起我之前的方法好很多了，之前我无法确定代码是否是有BUG，现在调试起来也不是那么困难，方便许多。

**Comment by hnwangkg-ezio** [comment]
你好，我有一个新的疑问，因为这块屏控制颜色需要一个short长度的数据，不过从TypeScript映射过来好像只能是number->int，难道我只能写一个枚举型，通过枚举型数据来在C++写一个对应的函数来控制吗？

**Comment by firemansamm** [comment]
从TS->C++用int，然后在C++强制类型转换 应该就可以了。

**Comment by hnwangkg-ezio** [comment]
这个我通过了其他的方法解决了，就是通过我上面的那种方法，这样我在TS中可以拖拽积木放置。
不过，我又遇到一个新的问题：
ezio@ubuntu:~/microbit/LED$ pxt test
Using target PXT/microbit with build engine yotta
  Target dir:   /home/ezio/microbit/node_modules/pxt-microbit
  PXT Core dir: /home/ezio/microbit/node_modules/pxt-core
Error: Assertion failed
    at Object.assert (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:54339:27)
    at /home/ezio/microbit/node_modules/pxt-core/built/pxt.js:63393:28
    at Array.forEach (native)
    at ProctoAssembler.emitRtCall (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:63392:32)
    at ProctoAssembler.emitExpr (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:63358:37)
    at ProctoAssembler.work (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:63154:34)
    at new ProctoAssembler (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:63089:22)
    at /home/ezio/microbit/node_modules/pxt-core/built/pxt.js:72083:27
    at Array.forEach (native)
    at serialize (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:72082:23)
    at Object.processorEmit (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:72192:23)
    at finalEmit (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:68176:26)
    at catchErrors (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:70294:33)
    at Object.compileBinary (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:68018:17)
    at Object.compile (/home/ezio/microbit/node_modules/pxt-core/built/pxt.js:70651:38)
    at /home/ezio/microbit/node_modules/pxt-core/built/pxt.js:82878:21
pxt_modules/core/dal.d.ts(1,1): error TS9200: Assertion failed

这种断言错误是指什么？
具体在我的
https://github.com/hnwangkg-ezio/lcd7
这个并没有提示那儿出了问题。

**Comment by hnwangkg-ezio** [comment]
具体是我的画线函数的问题

**Comment by hnwangkg-ezio** [comment]
这个断言好像是参数数量的问题。我减少了参数到四个，就编译通过了。奇怪
